Readme.md
# Project Corona

Mod Pack for Factorio.  Wrap the game in mods!

## News

[![GitHub release][badge-latest-release]][github-latest-release]
![GitHub Release Date][badge-release-date]

[Get it now][mod-portal-entry] on the mod portal, or in-game by searching for `Project Corona`.

### Problems & Suggestions

Please report any incompatibilities or problems on the [GitHub Issue Tracker][issue-tracker].  Suggestions are also welcome.

<!--
### Translations!

Place Holder
-->

### Latest

Added to the GUI that checks for duplicate custom inputs.

![GUI][readme-image-2]

### Previously

Added a GUI that has pre-loaded blueprint books.  All you have to do is press the button and have it inserted into your inventory.

![GUI][readme-image-1]

### Plans Going Forward

Add more optional dependencies and configure their quick keys to not conflict with one another by default.

## Description

The main objective is to have a list of optional dependencies that are tested for compatibility and play-ability.

This is an attempt to remove/replace duplicate or redundant items between multiple mods while adding some other things in.

~~No trees will spawn and no water tiles will be auto generated.~~ Added settings to make these optional in version 0.18.3.

Here are all the mods I use on my play-through's: [My Mod List][personal-mod-list]. (This probably won't be kept up to date, and some of these mods don't work with 0.18 unless you update their "factorio_version": to "0.18" in their info.json files)

#### Starting Items

Item|Qty
---|---:
Burner Mining Drill | 1
Burner Omnitractor (w/ Omnimatter mod) | 1
Iron Plate | 8
Magazine | 10
Pistol | 1
Seedling (w/ Bob's Greenhouse mod) | 50
Small Electric Pole (w/ Treeless Map enabled) | 20
Stone Furnace | 1
Stone Water Well (w/ Stone Water Well mod) | 1
Wood (removed w/ Treeless Map enabled) | 1

### Added/Manipulated Items

Adds an MK6 Power Armor if [Bob's Warfare Mod][bobwarfare-mod-page] is used.

Re-skins the [Miniloader Mod][miniloader-mod-page] with [Deadlock's Loaders][deadlock-mod-page] graphics.  This was done in order for them to match the look of the beltbox's correctly and will also load/unload train cargo wagons.

The code for "The Big Furnace" & "Circuit Factory" mods created by **[DellAquila][dellaquila-page]** has been used within this mod and doesn't require them as dependents. (Changed their size and removed their included loaders)

The ability to remove trees and water from spawning on the map.

## Contributors

DellAquila's [The Big Furnace Mod][the-big-furnace-mod]

DellAquila's [Circuit Factory Mod][circuit-factory-mod]

## Special Thanks

A lot of what I've learned to create this started with help from "[DWG][dwg-github-page]" the maintainer of "[Anoyomouse's][Anoyomouse-github-page]" [Warehousing Mod][warehousing-mod-page]

  [dellaquila-page]: https://mods.factorio.com/user/dellaquila
  [the-big-furnace-mod]: https://mods.factorio.com/mod/TheBigFurnace
  [circuit-factory-mod]: https://mods.factorio.com/mod/CircuitFactory
  
  [warehousing-mod-page]: https://mods.factorio.com/mod/Warehousing
  [dwg-github-page]: https://github.com/dgw
  [Anoyomouse-github-page]: https://github.com/Anoyomouse
  
  [bobwarfare-mod-page]: https://mods.factorio.com/mod/bobwarfare
  
  [miniloader-mod-page]: https://mods.factorio.com/mod/miniloader
  
  [deadlock-mod-page]: https://mods.factorio.com/mod/deadlock-beltboxes-loaders
  
  [mod-portal-entry]: https://mods.factorio.com/mod/project-corona
  [personal-mod-list]: https://github.com/Project-Corona/Project-Corona/blob/master/mod-list.txt
  <!--[forum-thread]:--> 
  
  [github-latest-release]: https://github.com/Project-Corona/Project-Corona/releases/latest
  [issue-tracker]: https://github.com/Project-Corona/Project-Corona/issues
   
  [badge-latest-release]: https://img.shields.io/github/v/release/project-corona/project-corona.svg?label=current+version
  [badge-release-date]: https://img.shields.io/github/release-date/project-corona/project-corona.svg?label=released
  
  [readme-image-1]: https://raw.githubusercontent.com/Project-Corona/Project-Corona/master/gui-image-1.jpg
  [readme-image-2]: https://raw.githubusercontent.com/Project-Corona/Project-Corona/master/gui-image-2.jpg


changelog.txt
Version: 1.0.0
Date: 2020-08-16

  Changed:
    - Internal code cleanup
    - Fixed a typo for boblogistics setting callout
    - Factorio version 1.0.0 Update
    - Removed freeplay cutscene so character spawns with correct items
---------------------------------------------------------------------------------------------------
Version: 0.18.10
Date: 2020-07-06

  Changed:
    - Added a button to the second tab of the GUI to find duplicate custom inputs.
---------------------------------------------------------------------------------------------------
Version: 0.18.9
Date: 2020-06-29

  Bug fix:
    - Fixed new games not spawning items correctly.
---------------------------------------------------------------------------------------------------
Version: 0.18.8
Date: 2020-06-28

  Changed:
    - Updated recipes
    - Updated Map Gen Presets to spawn omnimatter more like rail world maps.
    - Added a setting to change the size of the starting area to push enemies back if using
      mods that make enemies more difficult or cause more frequent attacks.
    - Hid some settings that don't have any effect when using omnimatter mod.
    - Added GUI - Currently only has a few pre-loaded blueprint books.
  Bug fix:
    - Fixed an issue causing the mod not to load if beltlayer recipe tweak was enabled and
      bob's basic belts were enabled.  (Finally I hope)
---------------------------------------------------------------------------------------------------
Version: 0.18.7
Date: 2020-04-23

  Changed:
    - Re-worked Loader technology prerequisites
    - Used miniloader mod's basic miniloader so it will now require electricity.
---------------------------------------------------------------------------------------------------
Version: 0.18.6
Date: 2020-04-23

  Bug fix:
    - Fixed an issue causing the mod not to load if beltlayer mod was not used
---------------------------------------------------------------------------------------------------
Version: 0.18.5
Date: 2020-04-20

  Changed:
    - Added a setting to allow beltlayer connector recipes to be changed to match normal
      underground belt recipes.
  Bug fix:
    - Fixed an issue causing a function to index a nil value.
    - Fixed miniloaders calling for a non-craftable recipe.
---------------------------------------------------------------------------------------------------
Version: 0.18.4
Date: 2020-04-20

  Bug fix:
    - Updated dependencies list to use latest versions of miniloader mod and deadlock's loaders
      to fix them not overwriting properly causing a non-craftable recipe to appear.
---------------------------------------------------------------------------------------------------
Version: 0.18.3
Date: 2020-04-16

  Changed:
    - Made all dependencies optional
    - Gave all the mod functions startup settings to pick and choose from
        - The Big Furnace
        - Circuit Factory
        - Loaderhaul
        - Power Armor MK6
        - Treeless Map
        - Waterless Map
    - Map Gen Presets will now work with Base, Bob's Ores, and Omnimatter or any combination of
      the three
---------------------------------------------------------------------------------------------------
Version: 0.18.2
Date: 2020-04-08

  Bug Fix:
    - Pipe connection on Circuit Factory now correctly located
---------------------------------------------------------------------------------------------------
Version: 0.18.1
Date: 2020-04-08

  Bug Fix:
    - Fixed recipe error caused by not having bobplates installed
---------------------------------------------------------------------------------------------------
Version: 0.18.0
Date: 2020-04-08

  Changed:
    - Initial Release
---------------------------------------------------------------------------------------------------


prototypes/entity/entity.lua
if settings.startup["enable-big-furnace"].value == true then
    data:extend({
        {
            type = "furnace",
            name = "big-furnace",
            icon = "__project-corona__/graphics/icons/big-furnace.png",
            icon_size = 32,
            flags = { "placeable-neutral", "placeable-player", "player-creation" },
            minable = { mining_time = 1, result = "big-furnace" },
            max_health = 350,
            corpse = "big-remnants",
            dying_explosion = "medium-explosion",
            resistances =
            {
                {
                    type = "fire",
                    percent = 80
                }
            },
            collision_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            selection_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            drawing_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            module_specification =
            {
                module_slots = 6,
                module_info_icon_shift = { 0, 0.8 }
            },
            allowed_effects = { "consumption", "speed", "productivity", "pollution" },
            crafting_categories = { "big-smelting" },
            result_inventory_size = 2,
            crafting_speed = 40,
            energy_usage = "3600kW",
            source_inventory_size = 1,
            energy_source =
            {
                type = "electric",
                usage_priority = "secondary-input",
                emissions = 0.005
            },
            vehicle_impact_sound = { filename = "__base__/sound/car-metal-impact.ogg", volume = 0.65 },
            working_sound =
            {
                sound =
                {
                    filename = "__base__/sound/electric-furnace.ogg",
                    volume = 0.7
                },
                apparent_volume = 1.5
            },
            animation =
            {
                layers =
                {
                    {
                        filename = "__project-corona__/graphics/entity/18x18-alignment.png",
                        priority = "high",
                        width = 288,
                        height = 288,
                        frame_count = 1,
                        shift = { 0, 0 },
                        scale = 2,
                        hr_version = {
                            filename = "__project-corona__/graphics/entity/hr-18x18-alignment.png",
                            priority = "high",
                            width = 576,
                            height = 576,
                            frame_count = 1,
                            shift = { 0, 0 },
                            scale = 1,
                        }
                    },
                    {
                        filename = "__base__/graphics/entity/electric-furnace/electric-furnace-base.png",
                        priority = "high",
                        width = 129,
                        height = 100,
                        frame_count = 1,
                        shift = { 2.5, 0 },
                        scale = 5.8,
                        hr_version =
                        {
                            filename = "__base__/graphics/entity/electric-furnace/hr-electric-furnace.png",
                            priority = "high",
                            width = 239,
                            height = 219,
                            frame_count = 1,
                            shift = { 0, 1 },
                            scale = 2.9,
                        }
                    },
                    {
                        filename = "__base__/graphics/entity/electric-furnace/electric-furnace-shadow.png",
                        priority = "high",
                        width = 129,
                        height = 100,
                        frame_count = 1,
                        scale = 5.8,
                        draw_as_shadow = true,
                        hr_version =
                        {
                            filename = "__base__/graphics/entity/electric-furnace/hr-electric-furnace-shadow.png",
                            priority = "high",
                            width = 227,
                            height = 171,
                            frame_count = 1,
                            draw_as_shadow = true,
                            shift = { 0, 0 },
                            scale = 2.9,
                        }
                    }
                }
            },
            working_visualisations =
            {
                {
                    animation =
                    {
                        filename = "__base__/graphics/entity/electric-furnace/electric-furnace-heater.png",
                        priority = "high",
                        width = 25,
                        height = 15,
                        frame_count = 12,
                        animation_speed = 0.5,
                        scale = 5.8,
                        shift = { 0, 5.5 },
                        hr_version =
                        {
                            filename = "__base__/graphics/entity/electric-furnace/hr-electric-furnace-heater.png",
                            priority = "high",
                            width = 60,
                            height = 56,
                            frame_count = 12,
                            animation_speed = 0.5,
                            shift = { 0.15, 5.9 },
                            scale = 2.79,
                        }
                    },
                    light = { intensity = 0.4, size = 6, shift = { 0.0, 1.0 }, color = { r = 1.0, g = 1.0, b = 1.0 } }
                },
                {
                    animation =
                    {
                        filename = "__base__/graphics/entity/electric-furnace/electric-furnace-propeller-1.png",
                        priority = "high",
                        width = 19,
                        height = 13,
                        frame_count = 4,
                        scale = 5.8,
                        animation_speed = 0.5,
                        shift = { -3.825, -3.7 },
                        hr_version =
                        {
                            filename = "__base__/graphics/entity/electric-furnace/hr-electric-furnace-propeller-1.png",
                            priority = "high",
                            width = 37,
                            height = 25,
                            frame_count = 4,
                            animation_speed = 0.5,
                            shift = { -3.8, -3.35 },
                            scale = 2.9,
                        }
                    }
                },
                {
                    animation =
                    {
                        filename = "__base__/graphics/entity/electric-furnace/electric-furnace-propeller-2.png",
                        priority = "high",
                        width = 12,
                        height = 9,
                        frame_count = 4,
                        scale = 5.55,
                        animation_speed = 0.5,
                        shift = { 0.39, -7.19 },
                        hr_version =
                        {
                            filename = "__base__/graphics/entity/electric-furnace/hr-electric-furnace-propeller-2.png",
                            priority = "high",
                            width = 23,
                            height = 15,
                            frame_count = 4,
                            animation_speed = 0.5,
                            shift = { 0.55, -6.9 },
                            scale = 2.9,
                        }
                    }
                }
            },
        },
    })
end

if settings.startup["enable-circuit-factory"].value == true then
    data:extend({
        {
            type = "assembling-machine",
            name = "circuit-factory",
            icon = "__project-corona__/graphics/icons/circuitfactory.png",
            icon_size = 32,
            flags = { "placeable-neutral", "placeable-player", "player-creation" },
            minable = { hardness = 0.2, mining_time = 0.5, result = "circuit-factory" },
            max_health = 400,
            corpse = "big-remnants",
            dying_explosion = "big-explosion",
            alert_icon_shift = util.by_pixel(-3, -12),
            resistances =
            {
                {
                    type = "fire",
                    percent = 70
                }
            },
            fluid_boxes =
            {
                {
                    production_type = "input",
                    pipe_picture = assembler3pipepictures(),
                    pipe_covers = pipecoverspictures(),
                    base_area = 10,
                    base_level = -1,
                    pipe_connections = { { type = "input", position = { 0.5, -9.5 } } },
                    secondary_draw_orders = { north = -1 }
                },
                {
                    production_type = "output",
                    pipe_picture = assembler3pipepictures(),
                    pipe_covers = pipecoverspictures(),
                    base_area = 10,
                    base_level = 1,
                    pipe_connections = { { type = "output", position = { 0.5, 9.5 } } },
                    secondary_draw_orders = { north = -1 }
                },
                off_when_no_fluid_recipe = true
            },
            open_sound = { filename = "__base__/sound/machine-open.ogg", volume = 0.85 },
            close_sound = { filename = "__base__/sound/machine-close.ogg", volume = 0.75 },
            vehicle_impact_sound = { filename = "__base__/sound/car-metal-impact.ogg", volume = 0.65 },
            working_sound =
            {
                sound = {
                    {
                        filename = "__base__/sound/assembling-machine-t3-1.ogg",
                        volume = 0.8
                    },
                    {
                        filename = "__base__/sound/assembling-machine-t3-2.ogg",
                        volume = 0.8
                    },
                },
                idle_sound = { filename = "__base__/sound/idle1.ogg", volume = 0.6 },
                apparent_volume = 1.5,
            },
            collision_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            selection_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            drawing_box = { { -9.0, -9.0 }, { 9.0, 9.0 } },
            animation =
            {
                layers =
                {
                    {
                        filename = "__project-corona__/graphics/entity/18x18-alignment.png",
                        priority = "high",
                        width = 288,
                        height = 288,
                        repeat_count = 32,
                        shift = { 0, 0 },
                        scale = 2,
                        hr_version = {
                            filename = "__project-corona__/graphics/entity/hr-18x18-alignment.png",
                            priority = "high",
                            width = 576,
                            height = 576,
                            repeat_count = 32,
                            shift = { 0, 0 },
                            scale = 1,
                        }
                    },
                    {
                        filename = "__base__/graphics/entity/assembling-machine-3/assembling-machine-3.png",
                        priority = "high",
                        width = 108,
                        height = 119,
                        frame_count = 32,
                        line_length = 8,
                        shift = util.by_pixel(0, -0.5),
                        scale = 6,
                        hr_version = {
                            filename = "__base__/graphics/entity/assembling-machine-3/hr-assembling-machine-3.png",
                            priority = "high",
                            width = 214,
                            height = 237,
                            frame_count = 32,
                            line_length = 8,
                            shift = util.by_pixel(0, -0.75),
                            scale = 3
                        }
                    },
                    {
                        filename = "__base__/graphics/entity/assembling-machine-3/assembling-machine-3-shadow.png",
                        priority = "high",
                        width = 130,
                        height = 82,
                        frame_count = 32,
                        line_length = 8,
                        scale = 6,
                        draw_as_shadow = true,
                        shift = util.by_pixel(28, 4),
                        hr_version = {
                            filename = "__base__/graphics/entity/assembling-machine-3/hr-assembling-machine-3-shadow.png",
                            priority = "high",
                            width = 260,
                            height = 162,
                            frame_count = 32,
                            line_length = 8,
                            draw_as_shadow = true,
                            shift = util.by_pixel(28, 4),
                            scale = 3
                        }
                    },
                },
            },
            crafting_categories = { "circuits" },
            crafting_speed = 3,
            energy_source =
            {
                type = "electric",
                usage_priority = "secondary-input",
                emissions = 0.03 / 3.5
            },
            energy_usage = "1000kW",
            ingredient_count = 10,
            module_specification =
            {
                module_slots = 5
            },
            allowed_effects = { "consumption", "speed", "productivity", "pollution" }
        },
    })
end
